File: apps/java-sdk/src/main/java/com/firecrawl/models/AgentListResponse.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;

/**
 * Response from listing agent runs.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AgentListResponse {

    private boolean success;
    private List<AgentListItem> agents;
    /** Absolute URL of the next page; only present when more pages exist. */
    private String next;
    private String error;

    public boolean isSuccess() { return success; }
    public List<AgentListItem> getAgents() { return agents; }
    public String getNext() { return next; }
    public String getError() { return error; }

    @Override
    public String toString() {
        int count = agents != null ? agents.size() : 0;
        return "AgentListResponse{success=" + success + ", agents=" + count + "}";
    }
}

File: apps/php-sdk/src/Models/AgentListResponse.php
<?php

declare(strict_types=1);

namespace Firecrawl\Models;

final class AgentListResponse
{
    /**
     * @param list<AgentListItem> $agents
     */
    public function __construct(
        private readonly bool $success = false,
        private readonly array $agents = [],
        private readonly ?string $next = null,
        private readonly ?string $error = null,
    ) {}

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        $agents = [];
        $rawAgents = $data['agents'] ?? [];
        if (is_array($rawAgents)) {
            foreach ($rawAgents as $agent) {
                $agents[] = AgentListItem::fromArray($agent);
            }
        }

        return new self(
            success: (bool) ($data['success'] ?? false),
            agents: $agents,
            next: $data['next'] ?? null,
            error: $data['error'] ?? null,
        );
    }

    public function isSuccess(): bool
    {
        return $this->success;
    }

    /** @return list<AgentListItem> */
    public function getAgents(): array
    {
        return $this->agents;
    }

    /**
     * Absolute URL of the next page; only present when more pages exist.
     */
    public function getNext(): ?string
    {
        return $this->next;
    }

    public function getError(): ?string
    {
        return $this->error;
    }
}
